### PR TITLE
chmod: move from getopts to clap

### PR DESCRIPTION
**Pull request by jhscheer**

### CODE REVIEW

File: src/uu/chmod/src/chmod.rs [hunk 1]
```diff
@@ -18,115 +19,173 @@ use uucore::fs::display_permissions_unix;
...
     }
 
     0
 }
 
-fn sanitize_input(args: &mut Vec<String>) -> Option<String> {
+// Iterate 'args' and delete the first occurrence
+// of a prefix '-' if it's associated with MODE
+// e.g. "chmod -v -xw -R FILE" -> "chmod -v xw -R FILE"
+fn strip_minus_from_mode(args: &mut Vec<String>) -> bool {
```
Comment (sylvestre) [inline src/uu/chmod/src/chmod.rs:173]: this function would need some unit test at the function level

**Review by sylvestre — CHANGES_REQUESTED** [review]
would be nice to have tests
like
chmod -x file
or
chmod 755 file
or
chmod g+r file

File: src/uu/chmod/src/chmod.rs (hunk 1 above)
Comment (jhscheer) [reply src/uu/chmod/src/chmod.rs]: done

**Comment by jhscheer** [comment]
Your suggestions are already covered by the numerous existing tests, e.g. in `test_chmod_ugoa`, `test_chmod_preserve_root`, etc.

**Comment by sylvestre** [comment]
Great and well done !

**Pull request merged**